src/server/checker/CheckerService.ts

```diff
@@ -68,19 +79,79 @@ export class CheckerService {
     id,
     user
   ) => {
-    const result = await this.CheckerModel.findByPk(
-      id,
-      user ? { include: [this.UserModel] } : undefined
-    )
-    if (result && user) {
-      const isAuthorized = result.users?.some((u) => u.id === user.id)
-      if (!isAuthorized) {
-        const resultWithoutUsers: Checker = { ...(result.toJSON() as Checker) }
-        delete resultWithoutUsers.users
-        return resultWithoutUsers
+    return await this.findAndCheckAuth(id, user)
+  }
+
+  retrievePublished: (
+    id: string
+  ) => Promise<GetPublishedCheckerWithoutDraftCheckerDTO | undefined> = async (
+    id
+  ) => {
+    const result = await this.PublishedCheckerModel.findOne({
```

Comment:
```suggestion
    const result = await this.PublishedCheckerModel.findOne({
        where: { checkerId: id },
        order: [['createdAt', 'DESC']],
     })
```

Would this work? This way we don't have to manually remove included `checker` later on.

Reply:
Making a commit with your suggestion because it is cleaner.

I also realized that we still needed to rename `checkerId` to `id` before returning the DB object. We can resolve this problem using the `attributes` option in the `findOne` method and rename `checkerId` to `id`.

Fixed in f9acf1f

```
    const result = await this.PublishedCheckerModel.findOne({
      attributes: [
        ['checkerId', 'id'], // rename checkerId as id 
        'title',
        'description',
        'fields',
        'constants',
        'operations',
        'displays',
      ],
      where: { checkerId: id },
      order: [['createdAt', 'DESC']],
    })

```

Reply:
Fixed in 0882214